screens/CheckoutScreen.js

```diff
@@ -141,111 +136,58 @@ export default class CheckoutScreen extends React.Component {
     ]);
-  }
-
-  applyReward() {
-    // Cannot apply reward if total price is less than 5
-    if (this.state.totalPrice < 5) {
-      return;
-    }
-    this.setState(prevState => ({
-      rewardsApplied: prevState.rewardsApplied + 1,
-      rewardsAvailable: prevState.rewardsAvailable - 1,
-      totalPrice: prevState.totalPrice - 5
-    }));
-  }
-
-  removeReward() {
-    this.setState(prevState => ({
-      rewardsApplied: prevState.rewardsApplied - 1,
-      rewardsAvailable: prevState.rewardsAvailable + 1,
-      totalPrice: prevState.totalPrice + 5
-    }));
-  }
-
-  // Generates rewards available as a list of TouchableOpacitys to display on checkout screen.
-  generateRewardsAvailable() {
-    const rewardsAvailable = [];
-    for (let i = 0; i < this.state.rewardsAvailable; i += 1) {
-      rewardsAvailable.push(
-        <TouchableOpacity key={i} onPress={() => this.applyReward()}>
-          <Text>APPLY $5 REWARD</Text>
-        </TouchableOpacity>
-      );
-    }
-    return rewardsAvailable;
-  }
-
-  // Generates rewards applied as a list of TouchableOpacitys to display on checkout screen.
-  generateRewardsApplied() {
-    const rewardsApplied = [];
-    for (let i = 0; i < this.state.rewardsApplied; i += 1) {
-      rewardsApplied.push(
-        <TouchableOpacity key={i} onPress={() => this.removeReward()}>
-          <Text>$5 REWARD APPLIED</Text>
-        </TouchableOpacity>
-      );
-    }
-    return rewardsApplied;
-  }
+  };
 
   render() {
     if (this.state.isLoading) {
       return null; // TODO @tommypoa waiting (flavicon?)
     }
 
-    const { cart, customer, products, totalPrice } = this.state;
+    const { cart, customer, totalPrice } = this.state;
 
     return (
-      // Temp fix for the horizontal orientation not showing Checkout Button
-      <ScrollView>
+      <View>
         <TopBar>
           <BackButton navigation={this.props.navigation} light={false} style={{ marginTop: 3, marginLeft: 24 }} />
           <Title> {'Customer: '.concat(customer.name)} </Title>
           {/* Duplicate, invisible element to have left-aligned BackButton */}
           <BackButton navigation={this.props.navigation} light={false} style={{ opacity: 0.0, disabled: true }} />
         </TopBar>
-        <View style={{ display: 'flex', flexDirection: 'row' }}>
+        <View style={{ display: 'flex', flexDirection: 'row', justifyContent: 'space-around' }}>
           {/* Display products */}
           <ProductsContainer>
-            <FlatListContainer
-              keyExtractor={product => product.id}
-              numColumns={5}
-              data={products}
-              renderItem={({ item }) => (
-                <TouchableOpacity onPress={() => this.addToCart(item)}>
-                  <ProductDisplayCard product={item} />
-                </TouchableOpacity>
-              )}
-            />
+            {Object.entries(cart).map(([id, product]) => (
```

Comment:
This is not a genuine nit since it’s so small, just a thought out loud. Since this mapping is used twice, a decent opportunity to create a functional component that takes in `cart` and `isLineItem`. 

Again, not actually recommending cuz it’s an annoying inconsequential change, but it makes me wonder if there is a future in React with a ton of mini functional components.....i don’t like functional components 😅😂

A middle ground I at some point learned (that Ethan hates) is the idea of creating a function along the lines of: 
```
renderProductList(cart, isLineItem) { 
  return Object.entries(cart)...
}
```
 
And then calling `this.renderProductList(cart, true)`. It’s like a mini functional component that lives in the same file? Yeah idk what my thoughts are on it

Reply:
HAHAHA. I'm totally going to do this because I think it's kind of unreadable as it is right now 🤠